tasks/src/lib.rs

```diff
@@ -80,22 +135,114 @@ pub trait Task: 'static {
 }
 
 /// Scheduler managing enqueued tasks and routing incoming responses to active tasks.
+#[derive(Debug)]
 pub struct Scheduler {
     flow: ClientFlow,
     waiting_tasks: TaskMap,
     active_tasks: TaskMap,
     tag_generator: TagGenerator,
+    capabilities: Vec1<Capability<'static>>,
```

Comment:
The inner capabilities. They are updated during the flow (during greetings, before auth, after auth etc). It also allows users to ask for capability availability (see function `ext_<capability>_supported()` functions below.